src/LineageTree/lineageTree.py

```diff
@@ -2533,66 +1821,97 @@ def unordered_tree_edit_distance(
+            k: v for k, v in hier.items() if k not in selected_cells
+        }
+        hier_selected = {k: v for k, v in hier.items() if k in selected_cells}
+        unselected = np.array(tuple(hier_unselected.values()))
+        x = []
+        y = []
+        if hier_unselected:
+            for pred, succs in lnks_tms["links"].items():
+                if pred not in selected_cells:
+                    for succ in succs:
+                        x.extend((hier[succ][0], hier[pred][0], None))
+                        y.extend((hier[succ][1], hier[pred][1], None))
+            ax.plot(x, y, c="black", linewidth=0.3, zorder=0.5, **kwargs)
+            ax.scatter(
+                unselected.T[0],
+                unselected.T[1],
+                s=0.1,
+                c="black",
+                zorder=1,
+                **kwargs,
+            )
+        if selected_cells:
+            selected = np.array(tuple(hier_selected.values()))
+            x = []
+            y = []
+            for pred, succs in lnks_tms["links"].items():
+                if pred in selected_cells:
+                    for succ in succs:
+                        x.extend((hier[succ][0], hier[pred][0], None))
+                        y.extend((hier[succ][1], hier[pred][1], None))
+            ax.plot(x, y, c=color, linewidth=0.3, zorder=0.4, **kwargs)
+            ax.scatter(
+                selected.T[0],
+                selected.T[1],
+                s=0.1,
+                c=color,
+                zorder=0.9,
+                **kwargs,
+            )
+        ax.get_yaxis().set_visible(False)
+        ax.get_xaxis().set_visible(False)
+        return ax
+
+    def to_simple_graph(self, node=None, start_time: int = 0):
         if node is None:
             mothers = [
                 root for root in self.roots if self.time[root] <= start_time
             ]
         else:
             mothers = node if isinstance(node, (list, set)) else [node]
-        graph = {}
-        all_nodes = {}
-        all_edges = {}
-        for mom in mothers:
-            edges = set()
-            nodes = set()
-            for branch in self.get_all_branches_of_node(mom):
-                nodes.update((branch[0], branch[-1]))
-                if len(branch) > 1:
-                    edges.add((branch[0], branch[-1]))
-                for suc in self[branch[-1]]:
-                    edges.add((branch[-1], suc))
-            all_edges[mom] = edges
-            all_nodes[mom] = nodes
-        for i, mother in enumerate(mothers):
-            graph[i] = nx.DiGraph()
-            graph[i].add_nodes_from(all_nodes[mother])
-            graph[i].add_edges_from(all_edges[mother])
-
-        return graph
+        return {
+            i: create_links_and_cycles(self, mother)
+            for i, mother in enumerate(mothers)
+        }
 
     def plot_all_lineages(
         self,
-        starting_point: int = 0,
+        nodes=[],
+        last_time_point_to_consider: int = 0,
```

Comment:
I realise that you have to change everywhere you put a default time point to `0`. It has to be something like
```python
def foo(tp: int | None = None):
    if tp is None:
        tp = self.t_b
```
Otherwise, the default behaviour will crash if you don't have a time point 0

Reply:
For napari I plot all the nodes that are long enough or have a label, do you think its better to do that than putting a timepoint?

Reply:
Yes, I think that's good.
That being said, that wasn't really the goal of my comment. Rather the fact that you are using `0` as default value for that function and many others (the "many others" here is important), which does not necessarily make sense for other dataset.

Reply:
Yes I know but I think it makes sense to plot all lineages when you call plot_all_lineages()

Reply:
Yes, the problem is that the default value should not be `0` but `self.t_b` because `0` is not always the first time point